src/factor_instances_provider/provider/provider_adopters/virtual_entity_creating_instance_provider.rs

```diff
@@ -0,0 +1,495 @@
+use crate::prelude::*;
+
+/// Uses a `FactorInstancesProvider` provide a VECI for a new virtual entity.
+pub struct VirtualEntityCreatingInstanceProvider;
+impl VirtualEntityCreatingInstanceProvider {
+    /// Uses a `FactorInstancesProvider` to provide a VECI for a new virtual entity.
+    ///
+    /// Reads FactorInstances for `factor_source` on `network_id` of kind `account_veci`,
+    /// meaning `(EntityKind::Account, KeyKind::TransactionSigning, KeySpace::Unsecurified)`,
+    /// from cache, if any, otherwise derives more of that kind AND other kinds:
+    /// identity_veci, account_mfa, identity_mfa
+    /// and saves into the cache and returns a collection of instances, split into
+    /// factor instance to use directly and factor instances which was cached, into
+    /// the mutable `cache` parameter.
+    ///
+    /// We are always reading from the beginning of each FactorInstance collection in the cache,
+    /// and we are always appending to the end.
+    pub async fn for_account_veci(
+        cache: &mut FactorInstancesCache,
+        profile: Option<Profile>,
+        factor_source: HDFactorSource,
```

Comment:
minor: would put `factor_source` & `profile` in the same order that we put `matrix_of_factor_sources` & `profile` on `SecurifyEntityFactorInstancesProvider`.

Applies to every fn inside this file

Reply:
I ensured to put cache first, then cache, then factor.

Functions which takes EntityKind takes those at as the very first. 

I think that order makes sense

Reply:
So the order is:
`cache`, `profile`, `factor_source`
vs
`cache`, `matrix_of_factor_sources`, `profile`



Given `matrix_of_factor_sources` is the securified version of one single `factor_source`, I would put them at same position among both providers.